_posts/2024-03-15-password-reset-vulnerability.md

```diff
@@ -0,0 +1,67 @@
+---
+title: Password Reset Vulnerabilities, Hacker One and Humility
+layout: post
+author: Martin Emde
+author_email: me@martinemde.com
+---
+
+Have you ever thrown actual spaghetti at a wall? It’s funny, sticky and barely induces any panic. HackerOne reports, on the other hand, have the opposite effect. Unlike wet spaghetti, the clean-up job is far more work for our security team.
+
+Running a bug bounty program means a stream of incoming reports, not all of them correct, that must be reviewed. After receiving enough dire-sounding reports that ultimately lead nowhere, it can look like thrown spaghetti (a see-what-sticks approach). Though we try to give each report a thorough, unbiased evaluation, it’s difficult to keep an open mind about any given report.
+
+Dead-end reports cost the RubyGems security team time, and slow down our ability to address more urgent security issues. I once spent days working on a vulnerability and the result was “clicking that checkbox in BurpSuite invalidates this approach.”
+
+But sometimes a hacker finds a very real security issue. This is a story about a recent bug report that I almost closed, assuming it was another false alarm, and how I realized I was wrong.
+
+## MFA bypass detected?
+
+On Dec 22nd, 2023, we received the report about multi-factor authentication (MFA) bypass on RubyGems.org’s password reset form. The report claimed to reset an MFA-protected user’s password using only the emailed password reset token, significantly undermining the value of MFA. Most reports tend to claim a similar severity level with varying accuracy. However, that’s not exactly what the report’s included screen capture looked like to me.
+
+In the screen recording, the reporter showed the following:
+
+1. Use a tool to store the full response from a valid password reset on an attacker controller user without MFA.
```

Comment:
I'm no fully sure, but should it be "an attacker controlled user without MFA", rather than "an attacker controller user without MFA"?

Reply:
You're right.